rust/arrow/src/datatypes.rs

```diff
@@ -1221,6 +1221,26 @@ impl Schema {
         &self.fields[i]
     }
 
+    /// Returns an immutable reference of a specific `Field` instance selected by name
+    pub fn field_with_name(&self, name: &str) -> Result<&Field> {
+        for i in 0..self.fields.len() {
```

Comment:
Why not use `index_of` inside `field_with_name`?

Reply:
Thanks @paddyhoran I implemented that suggestion.